compositor_api/src/types/register_output.rs

```diff
@@ -20,13 +20,21 @@ pub struct RtpOutputStream {
     pub ip: Option<Arc<str>>,
     /// (**default=`"udp"`**) Transport layer protocol that will be used to send RTP packets.
     pub transport_protocol: Option<TransportProtocol>,
-    pub video: Option<OutputRtpVideoOptions>,
-    pub audio: Option<OutputRtpAudioOptions>,
+    pub video: Option<OutputVideoOptions>,
+    pub audio: Option<OutputAudioOptions>,
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
 #[serde(deny_unknown_fields)]
-pub struct OutputRtpVideoOptions {
+pub struct Mp4Output {
+    pub path: String,
+    pub video: Option<OutputVideoOptions>,
+    pub audio: Option<OutputAudioOptions>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct OutputVideoOptions {
```

Comment:
You can't unify those values, after you add AAC here it will be incorrect for RTP. Start by duplicating and unifying when you know you can support unified interface everywhere.

Reply:
1. After implementing AAC I want to add support from AAC audio output in both: RTP and MP4
2. I purposefully left types refactoring for later. AAC encoder has a lot of options and I need to test which of them should be available.
3. Since writing opus audio into MP4 via FFmpeg didn't work and I could easily debug it, I want firstly implement AAC encoding, debug what's wrong / test if it works with AAC, and modify types later on (maybe writing opus into MP4 won't be available).